src/extension.ts

```diff
@@ -74,45 +78,59 @@ export function filterCatalogForRuntimes(catalog: any, runtimeIds:string[]) {
+async function generate(commandId: string, catalog: any) {
+    let telemetryProps: any = {
+        identifier: commandId,
+    };
+    const startTime = Date.now();
+    try {
+        let missions = catalog.missions.map((mission:any) => ({label: mission.id, description:mission.description}));
+        let missionId: any = await vscode.window.showQuickPick(missions, { placeHolder: 'Choose mission' });
+        if (missionId) {
+            let boosters = catalog.boosters.filter((item: any) => item.mission == missionId.label);
+            if (boosters) {
+                let runtimeId: any = await vscode.window.showQuickPick(boosters.map((item: any) => ({ label: item.runtime, description: item.version })), { placeHolder: 'Choose runtime' });
+                if (runtimeId) {
+                    let groupId = await vscode.window.showInputBox({ prompt: 'Group Id', placeHolder: 'Enter the group id', value: vscode.workspace.getConfiguration("project.initializer").get<string>("defaultGroupId") });
+                    if (groupId) {
+                        let artifactId = await vscode.window.showInputBox({ prompt: 'Artifact Id', placeHolder: 'Enter the artifact id', value: vscode.workspace.getConfiguration("project.initializer").get<string>("defaultArtifactId") });
+                        if (artifactId) {
+                            let version = await vscode.window.showInputBox({ prompt: 'Version', placeHolder: 'Enter the version', value: vscode.workspace.getConfiguration("project.initializer").get<string>("defaultVersion") });
+                            if (version) {
+                                vscode.window.setStatusBarMessage("Downloading zip project file", 1000);
+                                let zipProject = await catalogBuilder.zip(artifactId, missionId.label, runtimeId.label, runtimeId.description, groupId, artifactId, version);
+                                if (zipProject) {
+                                    let folder = await vscode.window.showWorkspaceFolderPick({ placeHolder: 'Select the target workspace folder' });
+                                    if (folder) {
+                                        vscode.window.setStatusBarMessage("Unzipping project file", 1000);
+                                        extract(zipProject as Buffer, folder.uri.fsPath);
+                                        vscode.window.showInformationMessage("Project saved to " + folder!.uri.fsPath);
+                                        telemetryProps.mission = missionId.label;
+                                        telemetryProps.runtime = runtimeId.label;
+                                        telemetryProps.runtimeVersion = runtimeId.description;
+                                    }
                                 }
                             }
                         }
                     }
                 }
             }
         }
+    } catch (error) {
+        telemetryProps.error = error.toString();
```

Comment:
make sure that user sensitive information cannot leak through exception message. For example `catalogBuilder.zip` method could fail and have local path reported as part of the exception's message.